### ISSUE TITLE
manhattan new york

### ISSUE THREAD

**Issue by missinglink** [openvenues/libpostal#53]
"manhattan new york" is incorrectly parsed as `city:new`, `state:york`.

would it be safe to assume that the word 'new' is always a prefix? (ie. a modifier for the following tokens)

```
> manhattan new york

Result:

{
  "state_district": "manhattan",
  "city": "new",
  "state": "york"
}

> wellington new zealand

Result:

{
  "house": "wellington",
  "country": "new zealand"
}
```

**Comment by thatdatabaseguy** [comment]
I wouldn't evaluate the parser on place queries at the moment as it's trained primarily on addresses. If you put a street address in front of that it'll parse to something decent. The new model I'm training will have many more examples of geocoder input like "manhattan new york" but those are relatively rare in the current training data because of the way component dropout works.

Heuristics for the word "new" aren't necessary. A sequence model can learn weights on context features that are effectively equivalent. In this case, "new york" is actually a known phrase, but the tagger currently doesn't require that the tokens within a phrase have the same label, primarily because the known phrase dictionaries are compiled from admin boundary names that occur > N times in the training data and some of the user-specified tags might be e.g. [addr:city=New York, NY](http://www.openstreetmap.org/node/2465888776). If that is used commonly enough to make it into the phrase dictionary, it can actually cause every instance of "new york ny" to be treated as a single phrase, which is not the desired effect.

In the new version I'll probably override user-specified boundary tags like addr:city with proper place names from OSM polygons, at least in cases where the user tag is composed of more than one admin component. That way the parser can safely treat multiword phrases as discrete tokens and avoid this entire class of mistakes.

**Comment by thatdatabaseguy** [comment]
The 1.0 release parses these and most other simple place queries correctly, as long as the toponyms occur somewhere in OSM, OpenAddresses, GeoPlanet, etc.

```
> manhattan new york

Result:

{
  "city_district": "manhattan",
  "city": "new york"
}

> Wellington New zealand

Result:

{
  "city": "wellington",
  "country": "new zealand"
}
```
